test/bin/mdx-test/expect/simple-mli/test-case.mli

```diff
@@ -19,14 +19,33 @@
       print_endline (String.concat " " [the; last; phrase])
     ]}
 
-    {[
+    With the optional header:
+
+    {@ocaml[
       # List.map (fun x -> x * x) [(1 + 9); 2; 3];;
       - : int list = [100; 4; 9]
       # List.map (fun x -> x * x) [1; 2; 3];;
       - : int list = [1; 4; 9]
     ]}
+
+    A shell block:
+
+    {@sh set-FOO=Hello,set-BAR=Bash[$ echo $FOO $BAR
+Hello Bash
+
+    ]}
```

Comment:
Thanks for pointing it out. It also adds a spurious newline where none should be.

After I looked into it it turns out that the whole mechanism of printing `Cram.t` is somewhat broken in MLI files in general, since it tries to compute padding but it doesn't actually get padded blocks. `Toplevel.t` blocks have special support for this so I adapted that. Overall it feels "dirty" that the code has so many `Syntax.Mli` switches because the behavior of the parsers in `.mli` and `.md` is apparently slightly different when it comes to whitespace and it requires lots of unpleasant "messing about" for lack of a better word.

We should probably adapt the Markdown parsing to match what odoc-parser outputs and just remove that handling. But maybe as a future PR.